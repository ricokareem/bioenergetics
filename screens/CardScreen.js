import React from 'react';
import PropTypes from 'prop-types';
import {
  FlatList,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import { ListItem } from 'react-native-elements';
import { Movies } from '../constants/MediaData';

const symptoms = [...Movies, { name: null, screen: null,  title: '' },];

const chevronProps = {
  color: '#fff',
  size: 28,
};

const CardScreen = (props) => {
  const { navigation } = props;
  // static navigationOptions = {
  //   title: 'Links',
  // };

  return (
    <View style={styles.container}>
      <FlatList
        data={symptoms}
        renderItem={({ item }) => (
          <TouchableOpacity
            style={styles.listItem}
            onPress={() => navigation.navigate(item.screen, {
              title: item.title,
              name: item.name,
              description: item.description,
              playlist: item.playlist
            })}
          >
            <Text style={styles.listText}>
              {item.title}
            </Text>
            <Text style={styles.listSubheadingText}>
              {item.subheading}
            </Text>
          </TouchableOpacity>
        )}
        numColumns={2}
        keyExtractor={(item, index) => index}
      />
    </View>
  );
};

CardScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
  }).isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  listItem: {
    backgroundColor: '#fc9829',
    flex: 1,
    flexDirection: 'column',
    margin: 1,
    padding: 20,
    height: 120,
  },
  listText: {
    fontSize: 12,
    fontWeight: 'bold',
    lineHeight: 14,
    color: '#fff',
  },
  listSubheadingText: {
    fontSize: 10,
    lineHeight: 12,
    color: '#fff',
  },
});

export default CardScreen;
