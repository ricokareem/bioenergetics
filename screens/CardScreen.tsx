import React from "react";
import {
  FlatList,
  ListRenderItemInfo,
  Route,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { Movies } from "../constants/MediaData";

type NavPropsType = {
  navigation: Route;
};

interface ItemType extends ListRenderItemInfo<any> {
  description: string;
  playlist: number[];
  subheading: string;
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    flex: 1,
  },
  listItem: {
    backgroundColor: "#fc9829",
    flex: 1,
    flexDirection: "column",
    height: 120,
    justifyContent: "center",
    margin: 1,
    padding: 20,
  },
  listText: {
    color: "#fff",
    fontSize: 12,
    fontWeight: "bold",
    lineHeight: 14,
  },
  listSubheadingText: {
    color: "#fff",
    fontSize: 10,
    lineHeight: 12,
    marginTop: 5,
  },
});

const symptoms = [...Movies, { name: null, screen: null, title: "" }];

const CardScreen: React.FC<NavPropsType> = (props) => {
  const { navigation } = props;

  return (
    <View style={styles.container}>
      <FlatList
        data={symptoms}
        renderItem={({ item }: ItemType) => (
          <TouchableOpacity
            style={styles.listItem}
            onPress={() => {
              navigation.navigate(item.screen, {
                title: item.title,
                name: item.name,
                description: item.description,
                playlist: item.playlist,
              });
            }}
          >
            <Text style={styles.listText}>{item.title}</Text>
            <Text style={styles.listSubheadingText}>{item.subheading}</Text>
          </TouchableOpacity>
        )}
        numColumns={2}
        keyExtractor={(item, index) => `${index}_${item.name}`}
      />
    </View>
  );
};

export default CardScreen;
