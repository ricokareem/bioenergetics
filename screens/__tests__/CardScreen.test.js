import React from 'react';
import { shallow } from 'enzyme';
import CardScreen from '../CardScreen';

describe('<CardScreen>', () => {
  it('should match to snapshot', () => {
    const wrapper = shallow(<CardScreen />);
    expect(wrapper).toMatchSnapshot();
  });
});
