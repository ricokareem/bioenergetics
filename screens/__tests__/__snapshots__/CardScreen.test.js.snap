// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardScreen> should match to snapshot 1`] = `ShallowWrapper {}`;
